src/common/components/formikWrappers/FormikDropdown.tsx

```diff
@@ -53,17 +53,23 @@ function FormikDropdown({
   const defaultValue = options.find((option) => option.value === value);
 
   return (
-    <Dropdown
-      {...field}
-      className={styles.formField}
-      defaultValue={defaultValue}
-      options={options}
-      onChange={handleChange}
-      invalid={Boolean(meta.error)}
-      helper={Boolean(meta.error) && t(meta.error || '')}
-      multiselect={false}
-      {...rest}
-    />
+    <div
+      onBlur={() => {
+        helpers.setTouched(true);
```

Comment:
This is pretty interesting to note actually.

https://reactjs.org/docs/events.html#focus-events

In vanilla JS, a div element would not have the ability to `blur`, but in React all elements do.

In your testing, how does this approach behave? Will `onBlur` be called whenever any element within this div loses focus? In that case the `Dropdown` would become touched after an item has been selected instead of after the focus leaves the component entirely.

They are also doing some changes to event handling in the upcoming versions of React
https://reactjs.org/blog/2020/08/10/react-v17-rc.html#aligning-with-browsers

They explicitly promise:
>Although React 17 switched from focus to focusin under the hood for the onFocus event, note that this has not affected the bubbling behavior. In React, onFocus event has always bubbled, and it continues to do so in React 17 because generally it is a more useful default. See this sandbox for the different checks you can add for different particular use cases.

I think it's implied that the same applies to `onBlur`, at least the sandbox has an example. So it's unlikely that they will remove this special behaviour.

Reply:
Good observation. I noticed now that the Invalid state is turned on when you activate the dropdown, and if you choose "select" and then select a value, it stays as invalid until you "tab" out of the dropdown. I'll see if I can improve the logic for invalid.

Reply:
I updated the PR with logic from that link, thanks a lot!